compiler/src/main/java/play/japi/twirl/compiler/TwirlCompiler.java

```diff
@@ -24,8 +24,9 @@ public class TwirlCompiler {
   public static final Set<String> DEFAULT_IMPORTS;
   static {
     Set<String> imports = new HashSet<>();
+    String scalaVersion = scala.util.Properties$.MODULE$.versionNumberString();
```

Comment:
I think `versionNumberString` is not useful.

```
Welcome to Scala 3.2.2 (11.0.19, Java OpenJDK 64-Bit Server VM).
Type in expressions for evaluation. Or try :help.
                                                                                                                                               
scala> scala.util.Properties.versionNumberString
val res0: String = 2.13.10
```

Reply:
oh...I didn't know this behavior.
Is there any other better way?

Reply:
@tototoshi On a second thought. I am not acquainted with usage of `TwirlCompiler.java`. If one uses Scala 3 and `TwirlCompiler.java` then I suppose there is the same issue with imports if `scala.util.Properties$.MODULE$.versionNumberString()` is not providing proper version for Scala 3.

Reply:
@ahjohannessen 
I understand that TwirlCompiler.java is an API designed for tools like Maven and Gradle.
I think the preferred behavior is that If you use TwirlCompiler.java of Scala3 build, it will output Scala code in the Scala3 syntax, and if you use a Scala2 build, it will output Scala2 code. However, when it comes to sbt 1.x, which always runs on Scala 2.12, there is a need to output code in the Scala3 syntax from Scala 2.12 builds

Reply:
I have no idea about a convenient method that can be a direct replacement for `versionNumberString`. However, if I place a class in src/main/scala-${version} that specifies the Scala major version and load it, I can determine whether it is Scala 2 or Scala 3."

Reply:
@tototoshi Found this on the internet:

```scala
object ScalaVersion extends App {
  def props(url: java.net.URL): java.util.Properties = {
    val properties = new java.util.Properties()
    val is = url.openStream()
    try {
      properties.load(is)
      properties
    } finally is.close()    
  }

  def scala2Version: String = 
    props(getClass.getResource("/library.properties")).getProperty("version.number")
    
  def checkScala3(res: java.util.Enumeration[java.net.URL]): String = 
    if (!res.hasMoreElements) scala2Version else {
      val manifest = props(res.nextElement)
      manifest.getProperty("Specification-Title") match {
        case "scala3-library-bootstrapped" =>
          manifest.getProperty("Implementation-Version")
        case _ => checkScala3(res)
      }
    }
  val manifests = getClass.getClassLoader.getResources("META-INF/MANIFEST.MF")
    
  val scalaVersion = checkScala3(manifests)
  val javaVersion = System.getProperty("java.version")

  println(s"Scala: $scalaVersion Java: $javaVersion")
}
```

Reply:
>  I place a class in src/main/scala-${version} that specifies the Scala major version and load it

👍 

> Found this on the internet:

I think this does not work if classpath include `scala3-library.jar` (e.g. use sbt `CrossVersion.for2_13Use3`)

Reply:
@xuwei-k I think this works:

```java
  public static final Set<String> DEFAULT_IMPORTS;
  static {
    Set<String> imports = new HashSet<>();
    String scalaVersion = play.twirl.compiler.BuildInfo$.MODULE$.scalaVersion();
    imports.addAll(JavaConverters$.MODULE$
        .seqAsJavaListConverter(play.twirl.compiler.TwirlCompiler$.MODULE$.defaultImports(scalaVersion)).asJava());
    DEFAULT_IMPORTS = Collections.unmodifiableSet(imports);
  }
```

and using sbt-buildinfo for compiler:

```scala
buildInfoKeys       := Seq[BuildInfoKey](scalaVersion),
buildInfoPackage := "play.twirl.compiler"
```

Reply:
@ahjohannessen Thanks! I will merge it.